salt/php-fpm/init.sls

```diff
@@ -22,7 +22,32 @@ php-fpm-reload:
     - name: service.reload
     - m_name: php{{ php_version }}-fpm.service
 
+/var/log/php-fpm:
+  file.directory:
+    - name: /var/log/php-fpm/
+    - makedirs: True
+
+/etc/logrotate.d/php-site-logs:
+  file.managed:
+    - name: /etc/logrotate.d/php-site-logs
+    - source: salt://logrotate/files/php-site-logs
+    - template: jinja
+    - context:
+        php_version: {{ php_version }}
+    - require:
+      - pkg: php-fpm
+      - file: /var/log/php-fpm
+
```

Comment:
We have this existing process for logrotate files: https://ocdsdeploy.readthedocs.io/en/latest/develop/update/logs.html#logrotate

Maybe we just add the option for a context (php_version here) to be provided to that existing state?

I don't think we need to replicate the `require`, since if php-fpm, etc. aren't installed, then there won't be any php-fpm log files to match.

Reply:
I've pushed a modification extending the logrotate state to support custom contexts.
If you're happy with this approach I will update kingfisher and the docs.

Reply:
PR looks good!